### PR TITLE
Add constraint slots so signal is passed through

### LINKED ISSUE #27180: Set constraint in simultaneous fit not working correctly
### Expected behavior
In any interface with simultaneous fitting if you add a function to fit and enter a constrain via the `...` button. The constraint should be added.  

### Actual behavior
The constraint is not added.

### Steps to reproduce the behavior
Go to `muon` -> `Frequency Domain Analysis`
load a workspace
go to the `fitting` tab and click `simultaneous fit`
Add a function e.g. `ExpDecay`
Click on the height value and click the `...` button, a dialog should open
Click set and go to `set constraint`.
Enter `0.1<Height<0.5`

If you right click on height, go to `constraints`->`10%`, this adds the constraint correctly. 

### Platforms affected
All

### PR DESCRIPTION
**Pull request by alicerussell1**
**Description of work.**
This PR fixes the add and remove constraint button in simultaneous fitting when adding via the `...` button.

**To test:**
Follow instructions in #27180 
Do the fit and the value should be between the constraints.

Fixes #27180 

*This does not require release notes* because **the bug did not exist in the last release**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by Matthew-Andrew — CHANGES_REQUESTED** [review]
Code changes look good and the interface now works for correct inputs. When incorrect strings are entered into the constraints box mantid crashes however.

**Review by Matthew-Andrew — CHANGES_REQUESTED** [review]
This now handles the has of grossly incorrect user input but will still crash if something like test<Height is put into the constraint box.

**Review by Matthew-Andrew — DISMISSED** [review]
This looks good :+1:

File: qt/widgets/common/src/FunctionBrowser/FunctionBrowserUtils.cpp [hunk 1]
```diff
@@ -105,7 +105,12 @@ splitConstraintString(const QString &constraint) {
         lowerBoundStr = QString::fromStdString(expr[1].str());
       }
       paramName = QString::fromStdString(expr[0].str());
-    } else {           // parameter is second
+    } else { // parameter is second
+      try {
```
Comment (DanNixon) [inline qt/widgets/common/src/FunctionBrowser/FunctionBrowserUtils.cpp]: `try_lexical_convert` would be a better option here.

Exception shouldn't be used for non-exceptional control flow.

**Review by DanNixon — CHANGES_REQUESTED** [review]

File: qt/widgets/common/src/FunctionBrowser/FunctionBrowserUtils.cpp (hunk 1 above)
Comment (alicerussell1) [reply qt/widgets/common/src/FunctionBrowser/FunctionBrowserUtils.cpp]: Ah yes that would be better - I didn't know that existed!

**Comment by alicerussell1** [comment]
@Mantid-Matthew There were an number of try catches in that method could be improves by using `try_lexical_convert` as suggested so I've changed them but can you test it still works as intended

**Comment by alicerussell1** [comment]
@DanNixon Changing to use `try_lexical_convert` seems to be failing the RHEL7 builds, it can't find it - do you know if it is in the version of boost used one RHEL?

**Comment by alicerussell1** [comment]
I've just checked and `try_lexical_convert` was added in version 1.56 and RHEL7/Ubuntu 16 is at 1.53 (although there is plans to upgrade it #26636) so I'm going to revert these changes for now so the bug can be fixed for this release.

**Comment by alicerussell1** [comment]
I have made an issue #27279 to use `try_lexical_convert` once boost has been updated

**Review by Matthew-Andrew — APPROVED** [review]
try_lexical_cast is the way to go but happy to improve this in the interim as is still an improvement over the hard crashes erroneous user input previously produced.

**Pull request merged**